components/governance/org.wso2.carbon.governance.lcm/src/main/java/org/wso2/carbon/governance/lcm/tasks/LCNotificationScheduler.java

```diff
@@ -0,0 +1,200 @@
+     *
+     * @param resource  resource which the lifecycle is attached.
+     * @param lifecycleName lifecycle name.
+     * @param tenantId  tenant Id.
+     * @param lifecycleState    new lifecycle state after the state changed.
+     * @throws GovernanceException
+     */
+    public void addScheduler(ResourceImpl resource, String lifecycleName, int tenantId, String lifecycleState)
+            throws GovernanceException {
+        if (resource != null && StringUtils.isNotEmpty(lifecycleName) && StringUtils.isNotEmpty(lifecycleState)) {
+
+            List checkpointsList = LifecycleStateDurationUtils
+                    .getDurationBeans(lifecycleName, lifecycleState);
+            if (checkpointsList != null) {
+                // Iterate through the checkpoint objects.
+                for (Object checkpoint : checkpointsList) {
+                    OMElement checkpointElement = (OMElement) checkpoint;
+                    OMElement boundary = checkpointElement.getFirstChildWithName(new QName
+                            (LifecycleConstants.LIFECYCLE_CONFIGURATION_NAMESPACE_URI, LifecycleConstants
+                                    .LIFECYCLE_CHECKPOINT_BOUNDARY));
+                    String minTimestamp = boundary
+                            .getAttribute(new QName(LifecycleConstants.LIFECYCLE_LOWER_BOUNDARY)).getAttributeValue();
+                    String maxTimestamp = boundary
+                            .getAttribute(new QName(LifecycleConstants.LIFECYCLE_UPPER_BOUNDARY)).getAttributeValue();
+
+                    long durationDifference = getDurationDifference(minTimestamp, maxTimestamp);
+
+                    // Set scheduler properties.
+                    LCNotification schedulerBean = new LCNotification();
+                    schedulerBean.setRegPath(resource.getPath());
+                    schedulerBean.setLcName(lifecycleName);
+                    schedulerBean.setUUID(resource.getUUID());
+                    schedulerBean.setTenantId(tenantId);
+                    schedulerBean.setLcCheckpointId(checkpointElement.getAttribute(new QName(checkpointId))
+                            .getAttributeValue());
+                    schedulerBean.setNotificationDate(getNotificationDate(durationDifference));
+
+                    LifecycleNotificationDAO scheduler = new JDBCLifecycleNotificationDAOImpl();
+                    try {
+                        scheduler.addScheduler(getRootRegistry(), schedulerBean);
+                    } catch (GovernanceException e) {
+                        log.error("Error while adding scheduler belongs to " + lifecycleName + "'s state '"
+                                + lifecycleState + "'to database", e);
+                    }
+                }
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.error("Invalid arguments supplied as lifecycle name: " + lifecycleName + ", lifecycle state: "
+                            + lifecycleState);
```

Comment:
It's a debug log. Hence I changed the error log to debug log. Thanks for pointing out.